pandas/core/arrays/integer.py

```diff
@@ -205,7 +205,14 @@ def coerce_to_array(values, dtype, mask=None, copy=False):
             mask = mask.copy()
         return values, mask
 
-    values = np.array(values, copy=copy)
+    if isinstance(values, list):
```

Comment:
How much more difficult would it be to do the proper fix as suggested by @jorisvandenbossche in the original issue?

Reply:
Not much more difficult I think. 
I assume we want to test if the `values` are array-like (so basically already have a `dtype`, i.e. numpy array, Series, Index, ), and in that case convert to a numpy array preserving the type. And otherwise always convert to object array.

Reply:
I am not fully sure how we do such a check in other places, maybe just actually checking for dtype? `hasattr(values, 'dtype')` ? Or explicitly checking for the different classes?

Reply:
Maybe that is better for now. Not sure if @jbrockmendel has come across this

Reply:
i think we usually check for a `dtype` attribute, but this might merit a helper function in case corner cases come up.  IIRC the last time I tried to write something general-ish I ended up with:

```
    if not hasattr(data, "dtype"):
        # e.g. list, tuple
        if np.ndim(data) == 0:
            # i.e. generator
            data = list(data)
        data = np.asarray(data)
```
https://github.com/pandas-dev/pandas/blob/master/pandas/core/arrays/datetimes.py#L1827
https://github.com/pandas-dev/pandas/blob/master/pandas/core/arrays/timedeltas.py#L991

Reply:
Tell me if there are any potential changes now?